webapp/src/pages/ResponsePage/ResponsePage.js

```diff
@@ -0,0 +1,211 @@
+                return ["unsubmitted" + " " + data.started_timestamp]
+            }
+            if (submitted) {
+                return ["submitted" + " " + data.submitted_timestamp]
+            }
+            if (withdrawn) {
+                return ["withdrawn" + " " + data.withdrawn_timestamp]
+            }
+        }
+    }
+
+
+
+    // Render Sections
+    renderSections() {
+        const applicationForm = this.state.applicationForm;
+        const applicationData = this.state.applicationData;
+        let html = [];
+        // Translation
+        const t = this.props.t;
+
+        // main function
+        if (applicationForm && applicationData) {
+            applicationForm.sections.forEach(section => {
+                html.push(<div key={section.name} className="section">
+                    { /*Heading*/}
+                    <div className="flex baseline"><h3>{section.name}</h3></div>
+                    { /*Q & A*/}
+                    <div className="Q-A">
+                        {this.renderQuestions(section)}
+                    </div>
+                </div>)
+            })
+        }
+
+        return html
+    }
+
+
+
+    // Render Questions 
+    renderQuestions(section) {
+        let questions = section.questions.map(q => {
+            return <div key={q.id} className="question-answer-block">
+                <p>{q.headline}</p>
+                <h6>{this.renderAnswer(q.id, q.type)}</h6>
+            </div>
+        })
+        return questions
+    }
+
+
+
+    // Render Answers 
+    renderAnswer(id, type) {
+        const applicationData = this.state.applicationData;
+        const baseUrl = process.env.REACT_APP_API_URL;
+        let answers;
+
+        applicationData.answers.forEach(a => {
+            if (a.question_id == id) {
+                formatAnswer(a, type)
+            }
+        })
+
+        // format aswerers 
+        function formatAnswer(a, type) {
+            // file
+            if (type == "file") {
+                answers = <a className="answer file" key={a.value} target="_blank" href={baseUrl + "/api/v1/file?filename=" + a.value}>{a.value}</a>
+            }
+            // multi-file
+            if (type == "multi-file") {
+                let files = [];
+                a.value.forEach((file => {
+                    file ? files.push(
+                        <div key={a.headline}><a key={a.headline} target="_blank" href={baseUrl + "/api/v1/file?filename=" + file} className="answer">{a.value}</a></div>
+                    )
+                        :
+                        console.log(`${a.question_id} contains no value`)
+                }))
+                answers = <div key={a.headline}>{files}</div>
+            }
+            // choice
+            if (type.includes("choice")) {
+                let choices = [];
+                a.options.forEach(opt => {
+                    a.value.forEach(val => {
+                        if (val.label == opt.label) {
```

Comment:
With the multi-choice question types, the `answer.value` will be a single string, not a list, and you need to match it to the options on the value not the label. So in your example JSON the answer with id 6 should look like this:

`{ "id": 6, "question_id": 44, "headline": "Third question", "value": "transport", "type": "multi-choice", "options": [{ "label": "Yes, transport only", "value": "transport" }, { "label": "Yes, accommodation only", "value": "accommodation" }] }`

Reply:
Please review to see if this is correct.